diff --git a/crfs/script/iptables_install.sh b/crfs/script/iptables_install.sh
--- a/crfs/script/iptables_install.sh
+++ b/crfs/script/iptables_install.sh
@@ -96,6 +96,9 @@
 		ln -s /usr/sbin/iptables-legacy-restore /usr/sbin/iptables-restore
 	fi
 
+	# tcp reuse
+  sysctl -w net.ipv4.tcp_tw_reuse=2
+
 	# Insert kernel modules for iptables and netfilter
 	insmod_when_not_loaded "x_tables.ko" || return 1
 	insmod_when_not_loaded "nf_conntrack.ko"
@@ -103,6 +106,67 @@
 	insmod_when_not_loaded "xt_state.ko"
 	insmod_when_not_loaded "xt_tcpudp.ko"
 
+  insmod_when_not_loaded "nfnetlink.ko"
+  insmod_when_not_loaded "nfnetlink_acct.ko"
+  insmod_when_not_loaded "nfnetlink_log.ko"
+  insmod_when_not_loaded "nfnetlink_queue.ko"
+  insmod_when_not_loaded "ip_set.ko"
+	insmod_when_not_loaded "ip_set_bitmap_ip.ko"
+	insmod_when_not_loaded "ip_set_bitmap_ipmac.ko"
+	insmod_when_not_loaded "ip_set_bitmap_port.ko"
+	insmod_when_not_loaded "ip_set_hash_ip.ko"
+  insmod_when_not_loaded "ip_set_hash_ipmark.ko"
+  insmod_when_not_loaded "ip_set_hash_ipport.ko"
+  insmod_when_not_loaded "ip_set_hash_ipportip.ko"
+  insmod_when_not_loaded "ip_set_hash_ipportnet.ko"
+  insmod_when_not_loaded "ip_set_hash_mac.ko"
+  insmod_when_not_loaded "ip_set_hash_net.ko"
+  insmod_when_not_loaded "ip_set_hash_netiface.ko"
+	insmod_when_not_loaded "ip_set_hash_netnet.ko"
+	insmod_when_not_loaded "ip_set_hash_netport.ko"
+  insmod_when_not_loaded "ip_set_hash_netportnet.ko"
+  insmod_when_not_loaded "ip_set_list_set.ko"
+  insmod_when_not_loaded "nf_conntrack_netlink.ko"
+  insmod_when_not_loaded "nf_nat.ko"
+  insmod_when_not_loaded "nf_nat_redirect.ko"
+  insmod_when_not_loaded "nf_tables.ko"
+  insmod_when_not_loaded "nf_tables_arp.ko"
+  insmod_when_not_loaded "nf_tables_netdev.ko"
+  insmod_when_not_loaded "tun.ko"
+  insmod_when_not_loaded "xt_CHECKSUM.ko"
+  insmod_when_not_loaded "xt_CT.ko"
+  insmod_when_not_loaded "xt_DSCP.ko"
+  insmod_when_not_loaded "xt_HL.ko"
+  insmod_when_not_loaded "xt_RATEEST.ko"
+  insmod_when_not_loaded "xt_REDIRECT.ko"
+  insmod_when_not_loaded "xt_TCPMSS.ko"
+  insmod_when_not_loaded "xt_TCPOPTSTRIP.ko"
+  insmod_when_not_loaded "xt_TRACE.ko"
+  insmod_when_not_loaded "xt_addrtype.ko"
+  insmod_when_not_loaded "xt_cgroup.ko"
+  insmod_when_not_loaded "xt_comment.ko"
+  insmod_when_not_loaded "xt_connbytes.ko"
+  insmod_when_not_loaded "xt_connlabel.ko"
+  insmod_when_not_loaded "xt_connlimit.ko"
+  insmod_when_not_loaded "xt_connmark.ko"
+  insmod_when_not_loaded "xt_conntrack.ko"
+  insmod_when_not_loaded "xt_dscp.ko"
+  insmod_when_not_loaded "xt_iprange.ko"
+  insmod_when_not_loaded "xt_limit.ko"
+  insmod_when_not_loaded "xt_mac.ko"
+  insmod_when_not_loaded "xt_mark.ko"
+  insmod_when_not_loaded "xt_multiport.ko"
+  insmod_when_not_loaded "xt_nat.ko"
+  insmod_when_not_loaded "xt_osf.ko"
+  insmod_when_not_loaded "xt_quota.ko"
+  insmod_when_not_loaded "xt_rateest.ko"
+  insmod_when_not_loaded "xt_recent.ko"
+  insmod_when_not_loaded "xt_set.ko"
+  insmod_when_not_loaded "xt_statistic.ko"
+  insmod_when_not_loaded "llc.ko"
+  insmod_when_not_loaded "stp.ko"
+
+
 	# Load recent, log, limit modules for brute force protection on ssh service.
 	insmod_when_not_loaded "xt_recent.ko" ip_list_tot=100 ip_pkt_list_tot=210
 	insmod_when_not_loaded "xt_LOG.ko"
@@ -110,6 +174,15 @@
 	insmod_when_not_loaded "xt_limit.ko"
 	return 0
 }
+
+iptables_common2_setup()
+{
+  insmod_when_not_loaded "bridge.ko"
+  insmod_when_not_loaded "br_netfilter.ko"
+  insmod_when_not_loaded "nf_tables_inet.ko"
+  insmod_when_not_loaded "xt_TPROXY.ko"
+  return 0
+}
 
 iptables_setup()
 {
@@ -119,6 +192,17 @@
 	insmod_no_check "nf_defrag_ipv4.ko"
 	insmod_no_check "nf_conntrack_ipv4.ko"
 
+	insmod_no_check "nf_defrag_ipv4.ko"
+	insmod_no_check "nf_dup_ipv4.ko"
+	insmod_no_check "nf_nat_ipv4.ko"
+	insmod_no_check "nf_nat_masquerade_ipv4.ko"
+	insmod_no_check "nf_reject_ipv4.ko"
+	insmod_no_check "nf_tables_ipv4.ko"
+	insmod_no_check "ipt_MASQUERADE.ko"
+	insmod_no_check "iptable_mangle.ko"
+	insmod_no_check "iptable_nat.ko"
+	insmod_no_check "iptable_raw.ko"
+
 	# Load log module for brute force protection on ssh service.
 	insmod_no_check "nf_log_ipv4.ko"
 	insmod_no_check "ipt_LOG.ko"
@@ -143,21 +227,36 @@
 	insmod_no_check "nf_defrag_ipv6.ko"
 	insmod_no_check "nf_conntrack_ipv6.ko"
 
+  insmod_no_check "ip6table_mangle.ko"
+  insmod_no_check "nf_defrag_ipv6.ko"
+  insmod_no_check "nf_dup_ipv6.ko"
+	insmod_no_check "nf_nat_ipv6.ko"
+	insmod_no_check "nf_nat_masquerade_ipv6.ko"
+	insmod_no_check "nf_reject_ipv6.ko"
+	insmod_no_check "nf_tables_ipv6.ko"
+
 	# Load log module for brute force protection on ssh service.
 	insmod_no_check "nf_log_ipv6.ko"
 	insmod_no_check "ip6t_LOG.ko"
 }
 
+# Force ipv6 install
+/usr/local/modules/script/ipv6_install.sh
+
 if [ "$1" == "ipv4" ]; then
 	iptables_common_setup
 	iptables_setup
+	ip6tables_setup
+	iptables_common2_setup
 elif [ "$1" == "ipv6" ]; then
 	iptables_common_setup
 	ip6tables_setup
+	iptables_common2_setup
 elif [ "$1" == "all" ]; then
 	iptables_common_setup
 	iptables_setup
 	ip6tables_setup
+	iptables_common2_setup
 else
 	echo "$0 : wrong parameter!"
 fi
